extensions/jaxb/deployment/src/main/java/io/quarkus/jaxb/deployment/JaxbProcessor.java

```diff
@@ -170,18 +172,73 @@ void processAnnotationsAndIndexFiles(
+                    proxyDefinitions.produce(new NativeImageProxyDefinitionBuildItem(className,
+                            "com.sun.xml.bind.v2.model.annotation.Locatable"));
                     addReflectiveClass(reflectiveClass, true, false, className);
                 });
 
+        proxyDefinitions
+                .produce(new NativeImageProxyDefinitionBuildItem("com.sun.xml.bind.marshaller.CharacterEscapeHandler"));
+        proxyDefinitions
+                .produce(new NativeImageProxyDefinitionBuildItem("com.sun.xml.txw2.output.CharacterEscapeHandler"));
+
+        proxyDefinitions.produce(new NativeImageProxyDefinitionBuildItem("com.sun.xml.txw2.TypedXmlWriter"));
+
         for (JaxbFileRootBuildItem i : fileRoots) {
-            try (Stream<Path> stream = iterateResources(i.getFileRoot())) {
+            try (Stream<Path> stream = iterateResources(applicationArchivesBuildItem, i.getFileRoot())) {
                 stream.filter(p -> p.getFileName().toString().equals("jaxb.index"))
                         .forEach(p1 -> handleJaxbFile(p1, resource, reflectiveClass));
             }
         }
     }
 
+    @BuildStep
+    void processTypedXmlWriters(
+            CombinedIndexBuildItem combinedIndexBuildItem,
+            BuildProducer<NativeImageProxyDefinitionBuildItem> proxyDefinitions) {
+        IndexView index = combinedIndexBuildItem.getIndex();
+        DotName typedXmlWriterDN = DotName.createSimple("com.sun.xml.txw2.TypedXmlWriter");
+        // getAllKnownDirectImplementors skip interface, so use own recursion
+        produceRecursiveProxies(index, typedXmlWriterDN)
+                .map(DotName::toString)
+                .map(NativeImageProxyDefinitionBuildItem::new)
+                .forEach(proxyDefinitions::produce);
+    }
+
+    @BuildStep
+    void seeAlso(CombinedIndexBuildItem combinedIndexBuildItem,
+            BuildProducer<ReflectiveClassBuildItem> reflectiveItems) {
+        IndexView index = combinedIndexBuildItem.getIndex();
+        for (AnnotationInstance xmlSeeAlsoAnn : index.getAnnotations(XML_SEE_ALSO)) {
+            AnnotationValue value = xmlSeeAlsoAnn.value();
+            Type[] types = value.asClassArray();
+            for (Type t : types) {
+                reflectiveItems.produce(new ReflectiveClassBuildItem(false, false, t.name().toString()));
+            }
+        }
+    }
+
+    public static boolean ModifierIsInterface(ClassInfo classinfo) {
+        return Modifier.isInterface(classinfo.flags());
+    }
+
+    Stream<DotName> produceRecursiveProxies(IndexView index,
+            DotName interfaceDN) {
+        Stream<DotName> itf = Stream.of(interfaceDN);
+        Stream<DotName> subItf = index.getKnownDirectImplementors(interfaceDN).stream()
+                .filter(JaxbProcessor::ModifierIsInterface)
+                .map(ClassInfo::name)
+                .flatMap((className) -> produceRecursiveProxies(index, className));
```

Comment:
We prefer to avoid using streams and lambdas because it is costly computation-wise. I'd suggest using a plain for-each here.

Reply:
hmm, I am lost I have switch from foreach to that for you and now you want that I go back to forEach.
Can you provide a little pseudo code about stream you want ?
I find very eleguante to have Steam in return

you mean something like:
```
//call
produceRecursiveProxies(index, typedXmlWriterDN, proxyDefinitions)
//function def
void produceRecursiveProxies(IndexView index, DotName interfaceDN, BuildProducer<NativeImageProxyDefinitionBuildItem> proxyDefinitions)
for (ClassInfo cls : index.getKnownDirectImplementors(interfaceDN)) {
    if(Modifier.isInterface(cls.flags()) {
        produceRecursiveProxies(index, cls.name());
        proxyDefinitions::produce(new NativeImageProxyDefinitionBuildItem(cls.name().toString()));
    }
}

```

Reply:
if we just use the list of class it disapear so I guess review is ok ? ^^